docs/developer-docs/latest/update-migration-guides/migration-guides/v4/plugin/update-folder-structure.md

```diff
@@ -0,0 +1,362 @@
+5. Create an `index.js` file for each content-type to export the schema and, optionally, lifecycle hooks:
+
+    ```jsx
+    // path: ./src/plugins/my-plugin/server/content-types/<content-type-name>/index.js
+
+    const schema = require("./schema.json");
+    const lifecycles = require("./lifecycles.js"); // optional
+
+    module.exports = {
+      schema,
+      lifecycles, // optional
+    };
+    ```
+
+6. Create a `server/content-types/index.js` file.
+7. In `server/content-types/index.js`, export all content-types and make sure the key of each content-type matches the `singularName` found in the `info` object of the content-type’s `schema.json` file:
+
+    ```json
+    // path: ./src/plugins/my-plugin/server/content-types/content-type-a/schema.json
+
+    "info": {
+      "singularName": "content-type-a", // kebab-case required
+      "pluralName": "content-type-as", // kebab-case required
+      "displayName": "Content-Type A",
+      "name": "Content-Type A",
+    };
+    ```
+
+    ```jsx
+    // path: ./src/plugins/my-plugin/server/content-types/index.js
+
+    "use strict";
+
+    const contentTypeA = require("./content-type-a");
+    const contentTypeB = require("./content-type-b");
+
+    module.exports = {
+      // Key names should match info.singularName key values found in corresponding schema.json files
+      "content-type-a": contentTypeA,
+      "content-type-b": contentTypeB,
+    };
+    ```
+
+:::note
+Converting Strapi v3 models to v4 content-types also requires updating getters and, optionally, relations (see [plugin back end migration documentation](/developer-docs/latest/update-migration-guides/migration-guides/v4/plugin/migrate-back-end.md)).
+:::
+
+### Create entry files
+
+::: callout
+Strapi v3 plugins use a strict folder structure convention.
+
+In Strapi v4, the folder structure for plugins is flexible. However, each plugin needs at least a `strapi-server.js` entry file and a `strapi-admin.js` entry file. The 2 entry files are used to take advantage of, respectively, the [Server API](/developer-docs/latest/developer-resources/plugin-api-reference/server.html) for the back end of the plugin and the [Admin Panel API](/developer-docs/latest/developer-resources/plugin-api-reference/admin-panel.html) for the front end of the plugin, if interactions with the admin panel are required.
+:::
+
+To update the plugin to Strapi v4:
+
+- Create the `strapi-server.js` back-end entry file at the root of the plugin folder. The file should require all necessary files and export the back-end plugin interface (see [migrating the back end of a plugin](/developer-docs/latest/update-migration-guides/migration-guides/v4/plugin/migrate-back-end.md)).
+
+  ::: details Example strapi-server.js entry file
+
+    ```jsx
+    // path: ./src/plugins/my-plugin/strapi-server.js
```

Comment:
We are crossing a lot of paths here, we show this entry file with a ton of stuff, but then we reference a folder structure that has an index.js and this method doesn't follow our CLI generation

Reply:
Good point here, I'll rethink this part a bit to make sure it's consistent with the plugin migration guide (and not necessarily with the CLI).